source/girgs/source/Generator.cpp

```diff
@@ -23,11 +23,13 @@ std::vector<double> generateWeights(int n, double ple, int weightSeed, bool para
     {
         const auto tid = omp_get_thread_num();
         auto gen = default_random_engine{weightSeed >= 0 ? (weightSeed+tid) : std::random_device()()};
-        auto dist = std::uniform_real_distribution<>{};
+        auto dist = std::uniform_real_distribution<>{1.0, std::pow(0.5*n, -ple + 1)};
```

Comment:
You cannot pull the +1 after dist(gen) into the multiplication like this.

Reply:
Let c = std::pow(0.5*n, -ple + 1) and x = dist(gen). Then previously we had    (c - 1) * x + 1 which was a uniform variate on the interval [(c - 1)*0 + 1, (c - 1)* 1+ 1) = [1, c). So it looks okay to me ?